spark/src/test/scala/org/apache/comet/parquet/ParquetReadSuite.scala

```diff
@@ -1233,7 +1233,9 @@ abstract class ParquetReadSuite extends CometTestBase {
 
             withParquetDataFrame(data, schema = Some(readSchema)) { df =>
               // TODO: validate with Spark 3.x and 'usingDataFusionParquetExec=true'
-              if (enableSchemaEvolution || usingDataSourceExec(conf)) {
+              if (enableSchemaEvolution || CometConf.COMET_NATIVE_SCAN_IMPL
+                  .get(conf)
+                  .equals(CometConf.SCAN_NATIVE_DATAFUSION)) {
```

Comment:
So schema evolution does not apply to Iceberg?

Reply:
the schemaEvolution flag applies only to CometScan (native_comet and native_iceberg_compat share that bit of code). CometNativeScan handles schema evolution differently and in this test always produces correct results instead of throwing an exception.
Note that native_iceberg_compat can match the behaviour of native_datafusion since it is the same implementation underneath but we end up throwing an exception because the code is shared.